File: chapter 06/recipe 6.5/DniNetComVisibility/DniNetComVisibilityObj.cs
using System;
using System.Runtime.InteropServices;

namespace DniNetComVisibility
{
	public interface IComVisibility
	{
		[ComVisible(false)]
		void Method1();
		void Method2();
		[ComVisible(false)]
		int Property1 { get;set;}
		int Property2 { get;set;}
	}

	[ClassInterface(ClassInterfaceType.None)]
	public class DniNetComVisibilityObj : IComVisibility	
	{
		public void Method1()
		{
		}

		public void Method2()
		{
		}

		public int Property1
		{
			get{return 0;}
			set{}
		}

		public int Property2
		{
			get{return 0;}
			set{}
		}

		protected void ProtectedMethod1()
		{
		}

		private void PrivateMethod1()
		{
		}

	}
}

File: chapter 06/recipe 6.5/DniNetComVisibilityVB/DniNetComVisibilityObj.vb
Imports System.Runtime.InteropServices

Public Interface IComVisibility
    <ComVisible(False)> _
    Sub Method1()
    Sub Method2()
    <ComVisible(False)> _
    Property Property1() As Integer
    Property Property2() As Integer
End Interface

<ClassInterface(ClassInterfaceType.None)> _
Public Class DniNetComVisibilityObj
    Implements IComVisibility

    Public Sub Method1() Implements IComVisibility.Method1

    End Sub

    Public Sub Method2() Implements IComVisibility.Method2

    End Sub

    Public Property Property1() As Integer _
            Implements IComVisibility.Property1
        Get
            Return 0
        End Get
        Set(ByVal value As Integer)

        End Set
    End Property

    Public Property Property2() As Integer _
            Implements IComVisibility.Property2
        Get
            Return 0
        End Get
        Set(ByVal value As Integer)

        End Set
    End Property

    Protected Sub ProtectedMethod1()

    End Sub

    Private Sub PrivateMethod()

    End Sub
End Class
